openjdk.test.jck/src/test.jck/net/adoptopenjdk/stf/Jck.java

```diff
@@ -0,0 +1,968 @@
+					xjcCmd = jckBase + File.separator + "solaris" + File.separator + "bin" + File.separator + "xjc.sh";
+					jxcCmd = jckBase + File.separator + "solaris" + File.separator + "bin" + File.separator + "schemagen.sh";
+					genCmd = jckBase + File.separator + "solaris" + File.separator + "bin" + File.separator + "wsgen.sh";
+					impCmd = jckBase + File.separator + "solaris" + File.separator + "bin" + File.separator + "wsimport.sh";
+			} else {
+					throw new StfException("Unknown platform:: " + platform);
+			}
+			
+			fileContent += "concurrency " + concurrency + ";\n";
+			fileContent += "timeoutfactor 1" + ";\n";							// All Devtools tests take less than 1h to finish.
+			
+			fileContent += "set jck.env.testPlatform.os " + testPlatform + ";\n";
+			fileContent += "set jck.env.devtools.testExecute.cmdAsString \"" + pathToJava + "\"" + ";\n";
+			fileContent += "set jck.env.devtools.refExecute.cmdAsFile \"" + pathToJava + "\"" + ";\n";
+			
+			fileContent += "set jck.env.devtools.scriptEnvVars \"" + "JAVA_HOME=\"" + testJdk + "\" TOOLS_HOME=\"" + testJdk + "\"" + "\"" + ";\n";
+			
+			if (tests.contains("java2schema")) {
+				fileContent += "set jck.env.devtools.jaxb.jxcCmd \"" + jxcCmd + "\"" + ";\n";
+			} else if (tests.contains("jaxws")) {
+				fileContent += "set jck.env.devtools.jaxws.cmdJavac \"" + pathToJavac + "\"" + ";\n";
+				fileContent += "set jck.env.devtools.jaxws.genCmd \"" + genCmd + "\"" + ";\n";
+				fileContent += "set jck.env.devtools.jaxws.impCmd \"" + impCmd + "\"" + ";\n";
+			} else {
+				fileContent += "set jck.env.devtools.jaxb.xjcCmd \"" + xjcCmd + "\"" + ";\n";
+			}
+
+			extraJvmOptions += suppressOutOfMemoryDumpOptions;
+			
+			// Add the JVM options supplied by the user plus those added in this method to the jtb file option.
+			fileContent += "set jck.env.devtools.refExecute.otherOpts \" " + extraJvmOptions + " \"" + ";\n";	
+		}
+	
+	}
+
+	public void jck9ConfigurationForMultijvmWithNoAgent(StfCoreExtension test) throws Exception {
+		String pathToJava = test.createJavaProcessDefinition().getCommand();
+		String pathToRmic = testJdk + File.separator + "bin" + File.separator + "rmic";
+		String pathToLib = testJdk + File.separator + "lib";
+		String pathToJavac = testJdk + File.separator + "bin" + File.separator + "javac";
+		DirectoryRef jckRuntimeNativeLibValue = nativesLoc;
+		DirectoryRef jckRuntimeJmxLibValue = nativesLoc;
+		String concurrency = "";
+		String keyword = "";
+		String libPath = "";
+		String testPlatform = "";
+		String robotAvailable = "";
+		String hostname = "";
+		String ipAddress = "";
+		extraJvmOptions = "";
+		
+		extraJvmOptions += " " + test.getJavaArgs(test.env().primaryJvm()) + " ";
+		
+		InetAddress addr = InetAddress.getLocalHost();
+        ipAddress = addr.getHostAddress();  
+        hostname = addr.getHostName();
+       
+        freePort = getFreePort();
+		
+		if (freePort == -1) {
+			throw new StfException("Unable to get a free port");
+		}
+		
+        
```

Comment:
Indentation could be fixed in the following block of if-else statements .

Reply:
Done